pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -362,6 +360,7 @@ public boolean addRecord(MutableSegment segment, RecordInfo recordInfo) {
     }
   }
 
+  // doAddRecord returns whether the event is out-of-order or not, if out-of-order returns false else true
```

Comment:
```suggestion
  /**
    Returns {@code true} when the record is added to the upsert metadata manager, {@code false} when the record is out-of-order thus not added.
  */
```

Reply:
We should add the same documentation to `PartitionUpsertMetadataManager.addRecord()`